library/src/scala/IArray.scala

```diff
@@ -260,59 +248,125 @@ object IArray:
+    def lengthCompare(len: Int): Int = genericArrayOps(arr).lengthCompare(len)
+    def padTo[U >: T: ClassTag](len: Int, elem: U): IArray[U] = genericArrayOps(arr).padTo(len, elem)
+    def partitionMap[T1: ClassTag, T2: ClassTag](f: T => Either[T1, T2]): (IArray[T1], IArray[T2]) = genericArrayOps(arr).partitionMap(f)
+    def patch[U >: T: ClassTag](from: Int, other: IterableOnce[U], replaced: Int): IArray[U] = genericArrayOps(arr).patch(from, other, replaced)
+    def permutations: Iterator[IArray[T]] = genericArrayOps(arr).permutations
+    def prepended[U >: T: ClassTag](x: U): IArray[U] = genericArrayOps(arr).prepended(x)
+    def prependedAll[U >: T: ClassTag](prefix: IterableOnce[U]): IArray[U] = genericArrayOps(arr).prependedAll(prefix)
+    def reverseIterator: Iterator[T] = genericArrayOps(arr).reverseIterator
+    def search[U >: T](elem: U)(using Ordering[U]): Searching.SearchResult = arr.toSeq.search(elem)
+    def search[U >: T](elem: U, from: Int, to: Int)(using Ordering[U]): Searching.SearchResult = arr.toSeq.search(elem, from, to)
+    def sizeCompare(that: IArray[Any]): Int = arr.toSeq.sizeCompare(that)
+    def sizeCompare(that: Iterable[_]): Int = arr.toSeq.sizeCompare(that)
+    def sizeCompare(otherSize: Int): Int = genericArrayOps(arr).sizeCompare(otherSize)
+    def sliding(size: Int, step: Int = 1): Iterator[IArray[T]] = genericArrayOps(arr).sliding(size, step)
+    def stepper[S <: Stepper[_]](using StepperShape[T, S]): S = genericArrayOps(arr).stepper[S]
+    def tails: Iterator[IArray[T]] = genericArrayOps(arr).tails
+    def tapEach[U](f: (T) => U): IArray[T] =
+      arr.toSeq.foreach(f)
+      arr
+    def unzip[T1, T2](using asPair: T => (T1, T2), ct1: ClassTag[T1], ct2: ClassTag[T2]): (IArray[T1], IArray[T2]) = genericArrayOps(arr).unzip
+    def unzip3[T1, T2, T3](using asTriple: T => (T1, T2, T3), ct1: ClassTag[T1], ct2: ClassTag[T2], ct3: ClassTag[T3]): (IArray[T1], IArray[T2], IArray[T3]) = genericArrayOps(arr).unzip3
+    def updated[U >: T: ClassTag](index: Int, elem: U): IArray[U] = genericArrayOps(arr).updated(index, elem)
+    def view: SeqView[T] = genericArrayOps(arr).view
+    def withFilter(p: T => Boolean): WithFilter[T] = new WithFilter(p, arr)
+    def zip[U](that: IArray[U]): IArray[(T, U)] = genericArrayOps(arr).zip(that)
+    def zip[U](that: IterableOnce[U]): IArray[(T, U)] = genericArrayOps(arr).zip(that)
+    def zipAll[T1 >: T, U](that: IArray[U], thisElem: T1, thatElem: U): IArray[(T1, U)] = genericArrayOps(arr).zipAll(that, thisElem, thatElem)
+    def zipAll[T1 >: T, U](that: Iterable[U], thisElem: T1, thatElem: U): IArray[(T1, U)] = genericArrayOps(arr).zipAll(that, thisElem, thatElem)
+    def zipWithIndex: IArray[(T, Int)] = genericArrayOps(arr).zipWithIndex
+  end extension
+
+  extension [T](arr: IArray[T])
+    def transpose[U](implicit asArray: T => IArray[U]): IArray[IArray[U]] =
+      genericArrayOps(arr).transpose(using asArray.asInstanceOf[T => Array[U]])
```

Comment:
Any reason this one is not in the big collective extension above?

Reply:
No.